File: ed2/gara4/ruspa/att/ruspa.c
#include <stdio.h>
#include <assert.h>

#define MAXN 100000

void inizia(int N, int M[]) {
    // Mettete qui il codice della soluzione
}

int abbatti(int X, int D) {
    // Mettete qui il codice della soluzione
    return D;
}


int M[MAXN];

int main() {
    FILE *fr, *fw;
    int N, T, X, D, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d %d", &N, &T));
    for(i=0; i<N; i++)
        assert(1 == fscanf(fr, "%d", &M[i]));

    inizia(N, M);
    for(i=0; i<T; i++) {
        if (i>0) fprintf(fw, " ");
        assert(2 == fscanf(fr, "%d %d", &X, &D));
        fprintf(fw, "%d", abbatti(X, D));
    }
    fprintf(fw, "\n");
    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed2/gara4/ruspa/att/ruspa.pas
procedure inizia(N: longint; var M: array of longint);
begin
    (* Mettete qui il codice della soluzione *)
end;

function abbatti(X, D: longint): longint;
begin
    (* Mettete qui il codice della soluzione *)
    abbatti := D;
end;


const
    MAXN = 100000;
var
    N, T, X, D, i   : longint;
    M               : array[0..MAXN-1] of longint;
    fr, fw          : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, T);
    for i:=0 to N-1 do
        read(fr, M[i]);
    readln(fr);

    inizia(N, M);
    for i:=0 to T-1 do begin
        if i > 0 then write(fw, ' ');
        readln(fr, X, D);
        write(fw, abbatti(X, D));
    end;
    writeln(fw);
    close(fr);
    close(fw);
end.
